lib/OpenQA/Scheduler/WorkerSlotPicker.pm

```diff
@@ -38,13 +48,22 @@ sub _reduce_matching_workers ($self) {
     return $picked_matching_worker_slots;
 }
 
+sub _id_or_skip ($self, $worker, $visited_worker_slots_by_id) {
+    my $id = $worker->id;
+    # skip slots that have already been picked
+    return undef if exists $visited_worker_slots_by_id->{$id};
+    # skip slots with "PARALLEL_ONE_HOST_ONLY" to try picking on any other hosts instead
+    return undef if $self->{_any_host} && $self->_is_one_host_only($worker);
+    return $id;
+}
+
 sub _pick_one_matching_slot_per_host($self, $job) {
     my $visited_worker_slots_by_id = $self->{_visited_worker_slots_by_id};
     my %visited_worker_slots_by_host;
     for my $matching_worker (@{$job->{matching_workers}}) {
-        my $id = $matching_worker->id;
-        next if exists $visited_worker_slots_by_id->{$id};    # skip slots that have already been picked
-        my $host = $matching_worker->host;
+        next unless my $id = $self->_id_or_skip($matching_worker, $visited_worker_slots_by_id);
+        next if $self->{_any_host} && $self->_is_one_host_only($matching_worker);
```

Comment:
Seems like this line is duplicated. the ` _id_or_skip` does the same check which will return `undef` and it will skip this line anyway. Probably it will never run.